primitives/account/src/logs.rs

```diff
@@ -0,0 +1,364 @@
+            Log::CreateStaker {
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::Stake {
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::UpdateStaker {
+                staker_address,
+                old_validator_address,
+                new_validator_address,
+            } => {
+                staker_address == address
+                    || old_validator_address
+                        .as_ref()
+                        .map(|old_address| old_address == address)
+                        .unwrap_or(false)
+                    || new_validator_address
+                        .as_ref()
+                        .map(|new_address| new_address == address)
+                        .unwrap_or(false)
+            }
+            Log::DeleteValidator {
+                validator_address,
+                reward_address,
+            } => validator_address == address || reward_address == address,
+            Log::Unstake {
+                staker_address,
+                validator_address,
+                ..
+            } => {
+                staker_address == address
+                    || validator_address
+                        .as_ref()
+                        .map(|validator_address| validator_address == address)
+                        .unwrap_or(false)
+            }
+            Log::PayoutReward { to, .. } => to == address,
+            Log::Park {
+                validator_address, ..
+            } => validator_address == address,
+            Log::Slash {
+                validator_address, ..
+            } => validator_address == address,
+            Log::RevertContract { contract_address } => contract_address == address,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+pub struct TransactionLog {
+    #[serde(rename = "hash")]
+    pub tx_hash: Blake2bHash,
+    pub logs: Vec<Log>,
+}
+
+impl TransactionLog {
+    pub fn new(tx_hash: Blake2bHash, logs: Vec<Log>) -> Self {
+        Self { tx_hash, logs }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde-derive", derive(serde::Serialize, serde::Deserialize))]
+// Renaming affects only the struct names and thus their tag, the "type" field.
```

Comment:
Mmm, it should affect all of the members according to the documentation: _Rename all the fields (if this is a struct) or variants (if this is an enum) according to the given case convention._

Reply:
It doesn't. It renames all _direct_ fields and variants.

Reply:
Ok, I think I now understand the point of the comment, maybe it should be reworded as:
_Renaming affects only the enum variant names and its tag (the "type" field)_
?